metadata-ingestion/src/datahub/ingestion/source/sql/oauth_generator.py

```diff
@@ -0,0 +1,63 @@
+import base64
+import logging
+from typing import Any, Dict, List, Union
+
+import msal
+from OpenSSL.crypto import FILETYPE_PEM, load_certificate
+
+log = logging.getLogger(__name__)
+
+
+class OauthTokenGenerator:
+    def __init__(self, client_id, authority_url):
+        self.client_id = client_id
+        self.authority_url = authority_url
+
+    def _get_token(
+        self,
+        credentials: Union[str, Dict[str, Any]],
+        scope: Union[str, List[str]],
+        check_cache: bool,
+    ) -> str:
+        app = msal.ConfidentialClientApplication(
```

Comment:
This makes it Microsoft specific. Can we please change the `OauthTokenGenerator` to have some options for future extensibility when the 2 other cloud providers support is added? Please keep this Microsoft specific caveat in mind while writing docs.

Reply:
added different functions for each provider, based on provider passed in the recipe
right now get_token function is only implemented for microsoft